pkg/kubelet/kuberuntime/kuberuntime_sandbox.go

```diff
@@ -228,6 +228,29 @@ func (m *kubeGenericRuntimeManager) generatePodSandboxWindowsConfig(pod *v1.Pod)
 		SecurityContext: &runtimeapi.WindowsSandboxSecurityContext{},
 	}
 
+	// If all of the containers in a pod are HostProcess containers, set the pod's HostProcess field
+	// explicitly because the container runtime requires this information at sandbox creation time.
+	if kubecontainer.HasWindowsHostProcessContainer(pod) {
+		// Pods containing HostProcess containers should fail to schedule if feature is not
+		// enabled instead of trying to schedule containers as regular containers as stated in
+		// PRR review.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.WindowsHostProcessContainers) {
```

Comment:
since it's breaking the previous behavior, this may break some customers on upgrade. Is there a possibility customer marked container as hostprocess and forgot to clean it up after?

Reply:
The situation where behavior should be changing is when users set `hostProcess=true` for only the containers, 
before this change the containers will start as normal containers, after the changes the containers will be started as `hostProcess` containers.

The expected behavior and behavior outlined in the KEP is that the container should be started as a `hostProcess` container.

We've had several people report this issue recently (including https://kubernetes.slack.com/archives/C0SJ4AFB7/p1652916420701559?thread_ts=1652890944.995179&cid=C0SJ4AFB7)

Reply:
talked offline. Seems like an edge case when things may break for users. Unless Im'm missing something this is not blocking the PR.

Reply:
If a customer thinks they are running in a Hostprocess pod but aren't, due to the limitations of process isolated containers, this will likely lead to functionality in the pod to not work in other ways leading them to see this as a bug. We have seen folks come to the sig-windows slack when this occurs.

An an example the promtheus windows_exporter, it technically runs inside a process isolated windows container but all the stats it [collects are wildly off and in some cases like querying for containers just don't work](https://github.com/prometheus-community/windows_exporter/pull/581#issuecomment-669849739). 

I would anticipate this would be the same for most of the use cases we envisioned where customers are modifing the host registry or interacting with host services.